core/sdk-core/src/main/java/software/amazon/awssdk/core/retry/conditions/OrRetryCondition.java

```diff
@@ -28,12 +28,16 @@
 @SdkPublicApi
 public final class OrRetryCondition implements RetryCondition {
 
-    private Set<RetryCondition> conditions = new HashSet<>();
+    private final List<RetryCondition> conditions = new ArrayList<>();
```

Comment:
Could this be a breaking change? Duplicate conditions provided would now get evaluated twice

Reply:
Good catch! I'll switch to a linked hash set, so that we can preserve order and uniqueness.

Reply:
I also went ahead and added a test to preserve ordering.